app/src/main/java/org/openobservatory/ooniprobe/model/database/Measurement.java

```diff
@@ -64,6 +67,25 @@ public Measurement(Result result, String test_name, String report_id) {
 		start_time = new java.util.Date();
 	}
 
+	public static Where<Measurement> selectUploadable() {
+		// We check on both the report_id and is_uploaded as we
+		// may have some unuploaded measurements which are marked
+		// as is_uploaded = true, but we always know that those with
+		// report_id set to null are not uploaded
+		return SQLite.select().from(Measurement.class)
+				.where(Measurement_Table.is_failed.eq(false))
+				.and(Measurement_Table.is_rerun.eq(false))
+				.and(Measurement_Table.is_done.eq(true))
+				.and(OperatorGroup.clause()
```

Comment:
Just to double check, did you make sure that `where(a, b, c) == where(a).and(b).and(c)`?

Reply:
Yes, confirmed